astropy/modeling/tests/test_quantities_fitting.py

```diff
@@ -111,7 +113,13 @@ def test_fitting_with_initial_values(fitter):
 
     # Fit the data using a Gaussian with units
     g_init = models.Gaussian1D(amplitude=1.0 * u.mJy, mean=3 * u.cm, stddev=2 * u.mm)
-    g = fitter(g_init, x, y)
+    # https://github.com/astropy/astropy/issues/16320 (scipy can be flaky in OSX ARM64)
+    if sys.platform == "darwin":
+        ctx = np.errstate(invalid="ignore")
+    else:
+        ctx = contextlib.nullcontext()
+    with ctx:
+        g = fitter(g_init, x, y)
```

Comment:
I'd like to avoid going with this patch if we can. The behavior isn't flaky, per say: it's deterministic. I'm working on reducing the reproducer outside of astropy so scipy folks can inspect what's going on more easily.

Reply:
I guess we can go with #16330 instead but I'd rather not have our CI red until scipy has time to fix it. Thanks for the quick feedback!